### PR TITLE
Fix undefined function description

### PR DESCRIPTION
**Pull request by digeff**
We got telemetry that description is sometimes undefined on formatFunctionDescription:
TypeError: Cannot read property 'replace' of undefined
    at formatFunctionDescription (index.js:251:10)
    at renderPreview (index.js:88:16)
    at renderObjectPreview (index.js:176:25)
    at renderPreview (index.js:91:16)
    at previewRemoteObjectInternal (index.js:61:11)
    at Object.previewRemoteObject (index.js:48:20)
    at _createObjectVariable (variables.js:417:31)
    at _createVariable (variables.js:388:25)
    at _createVariablesForProperty (variables.js:366:30)
    at _getObjectProperties (variables.js:276:37)"

### CODE REVIEW

**Comment by connor4312** [comment]
I think this should be fixed with https://github.com/microsoft/vscode-js-debug/commit/a629fed2d8fdd6c949f3a59c60bac04d1f576120#diff-ae55c9d7647ee09043b3ff02a6788f81

**Comment by digeff** [comment]
Sounds good! Closing

**Comment by cheechak** [comment]
I was just about to open a new bug on this when I found this issue.  I don't think the problem is fixed by that change.  In fact I think it may actually make it worse.

I've encountered this issue trying to get the vs code debugger to work when connecting to a custom CDP implementation.  Here's the exception callstack:
```
Error processing variables: TypeError: Cannot read property 'replace' of undefined
	at m (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2468767)
	at l (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2465644)
	at Object.t.previewRemoteObject (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2470391)
	at h._generateVariableValueDescription (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:43982)
	at h._createObjectVariable (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:43779)
	at h._createVariable (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:43249)
	at h._createVariablesForProperty (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:42547)
	at h._getObjectProperties (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:40868)
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:94:5)
	at async h.getVariables (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:37430)
	at async k._onVariables (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:73:125247)
	at async t.default._onMessage (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2463780)
```

It looks like the problems are caused by exceptions being thrown in `formatFunctionDescription()` (from src/adapter/objectPreview/index.ts) when trying to access an undefined description.

Basically all calls to this function are passing the "description" field of a [Runtime.RemoteObject](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#type-RemoteObject).  As you can see in the listing of the type in the protocol documentation, this field is supposed to be optional.

As it currently stands, if the target fails to provide this field the variables view basically becomes unusable.

**Comment by cheechak** [comment]
Actually it looks like this goes a bit further.  eg. If any object's Runtime.ObjectPreview doesn't provide a description field then line 158 of src/adapter/objectPreview/index.ts (in function `renderObjectPreview`) will cause an exception with a callstack like this:
```
Error processing variables: TypeError: Cannot read property 'length' of undefined
	at Object.t.trimEnd (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2493083)
	at /usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2466706
	at d (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2467658)
	at l (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2465747)
	at Object.t.previewRemoteObject (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2470391)
	at h._generateVariableValueDescription (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:43982)
	at h._createObjectVariable (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:43779)
	at h._createVariable (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:43249)
	at h._createVariablesForProperty (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:42547)
	at h._getObjectProperties (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:40868)
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:94:5)
	at async h.getVariables (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:48:37430)
	at async k._onVariables (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:73:125247)
	at async t.default._onMessage (/usr/share/code/resources/app/extensions/ms-vscode.js-debug/src/extension.js:15:2463780)
```

This case should easily be fixed by changing line 157 like so:
if (**preview.description &&** preview.description !== 'Object')

Interestingly, other functions like `renderPrimitivePreview` in the same file do check for and handle missing description fields, but there are a number of spots that don't.

**Comment by connor4312** [comment]
The types for remote objects are not well typed. Stronger typings [are used](https://github.com/microsoft/vscode-js-debug/blob/master/src/adapter/objectPreview/betterTypes.ts) as observed from Chrome itself. If your custom CDP implementation differs from what Chrome provides and causes errors, I would consider that a bug in your implementation rather than in the debugger.

**Comment by cheechak** [comment]
Someone else already encountered this issue elsewhere and reported it here before I commented on it.  Have you not considered that there may be edge cases where chrome may not report a description when you expect it (as the published spec explicitly allows for)?

Beyond that, what chrome / v8 / node provides could also change in the future.  It's good engineering practice to validate data received from an external source in any case.

Once I encountered this issue, I (not surprisingly) immediately worked around it, so it doesn't effect me.  I'm not suggesting this be changed for my sake.

I've posted some detailed information to try to help fix an edge case in the debugger where it can throw and ceases to work properly.  Responding to my effort to help by turning that around into accusation of me somehow writing buggy code comes off as a rather hostile response.
